torch/nn/_functions/thnn/auto_double_backwards.py

```diff
@@ -137,14 +236,39 @@ def nllloss_double_backwards(ctx, ggI):
     return gI, None, ggO, None, None, None
 
 
+def smoothl1loss_double_backwards(ctx, ggI):
+    size_average = ctx.additional_args[0]
+    input, target, gO = ctx.saved_variables
+    div_factor = input.nelement() if size_average else 1
+
+    input_sub_target = input - target
+    small_error_mask = (input_sub_target.abs() < 1)
+    large_error_mask = (small_error_mask == 0)
+    large_error_pos_mask = (((input_sub_target > 0) + large_error_mask) == 2).type_as(ggI)
```

Comment:
nit: it would be good at some point to add Variable support for `logical_and` etc.